### PR TITLE
Translation CLI

### LINKED ISSUE #72: CLI translation front-end
Add a `__main__.py` entry point script which can provide a simple command-line interface. Initially I think it would be good to add a translation sub-command which allows people to run zone translations in as simple a way as possible.

- [x] Create `__main__.py` with `argparser` setup to handle sub-commands
- [x] Add 'translate' sub-command in argparser to run zone translations

Translate sub-command would need the following parameters:
- data file: CSV file containing data to be translated
- translation file: path to CSV containing translation
- output: path to output file
- zone column: name of column containing zone ID, default 'zone_id'
- from column: name of column in translation containing from zone id, default 'from_id'
- to column: name of column in translation containing to zone id, default 'to_id'
- factor column: name of column in translation containing split factors, default 'split_factor'
- matrix: boolean flag for whether data file should be considered a matrix (if true zone column should be 2 values), default False

Given the number of parameters it might make more sense to have those be a config and just pass the path to a config as the argument?
Also it might make sense to split the sub-command into 2 separate ones one for vector translate and one for matrix translate, because matrix translate might want a format parameter to handle both wide and long formats.

### PR DESCRIPTION
**Pull request by wsp-mbuckley**
Describe Changes
---
Add CLI for running zone translations. Resolves #72 

Task Checklist
----
- [x] Have unittests been added (testing individual functions and their edge cases)
- [x] Have integration tests been added (if applicable, to test modules of functionality)
- [x] Updated RELEASE.md to reflect changes in PR
- [x] Have new dependencies been added?
  - [x] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [x] Have they been added to `pyproject.toml`

### CODE REVIEW

**Comment by wsp-mbuckley** [comment]
@BenTaylor-TfN & @isaac-tfn not sure which of you is best placed to review this (or has the time), but it's ready now.

File: src/caf/toolkit/__init__.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
 """A toolkit of transport planning and appraisal functionalities."""
+
 from ._version import __version__
 
+__homepage__ = "https://github.com/Transport-for-the-North/caf.toolkit"
+__source_url__ = "https://github.com/Transport-for-the-North/caf.toolkit"
```
[the remark this answers is not in the document]
Comment (wsp-mbuckley) [reply src/caf/toolkit/__init__.py:6]: They're used in the ToolDetails class for the logging, they're optional values but I thought it made sense to include them. They can be got rid of if needed or just moved to where the tool details is initialised in the `__main__` module

File: src/caf/toolkit/translation.py [hunk 2]
```diff
@@ -1060,8 +1071,226 @@ def pandas_multi_vector_zone_translation(
...
+        matrix,
+        lookup,
+        translation_from_col=from_col,
+        translation_to_col=to_col,
+        translation_factors_col=factors_col,
+    )
+
+    translated.index.name = matrix.index.name
+    translated.columns.name = matrix.columns.name
+    translated.stack().to_csv(output_path)
```
Comment (isaac-tfn) [inline src/caf/toolkit/translation.py:1295]: Could add in wide or long as options for the output?

**Review by isaac-tfn — COMMENTED** [review]
Overall looks pretty good, I've just added a couple of potential improvements. None of them should take long and this could be merged and released without them if time is short.

File: src/caf/toolkit/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenTaylor-TfN) [reply src/caf/toolkit/__init__.py:6]: I've got no strong opinion on this, happy to be left where is so they're at least easy to find if needed in future.

**Comment by BenTaylor-TfN** [comment]
> I've not added much in the way of documentation about the CLI but perhaps that may be worth adding in `__init__` module docstring and maybe more details in the `translation` module docstring, any thoughts on this?

Agreed that much of the selling point of this is going to be around how well documented and easy to use it is.

How does documentations usually work for CLI stuff? Is this something we will want to manually document within the documentation?
I'm thinking it's likely worth producing some files as some example input and output files to help people understand exactly how it works, the inputs can be a little confusing! Maybe stick these files in the repo under and "examples" folder, and direct people to there as part of the documentation?

File: src/caf/toolkit/__main__.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+            " file containing the zone ids to translate to"
+        },
+    )
+    factor_column: int | str = dataclasses.field(
+        default=2,
+        metadata={
+            "help": "The column (name or position) in the translation"
+            " file containing the weightings between from and to zones"
+        },
+    )
```
[the remark this answers is not in the document]
Comment (wsp-mbuckley) [reply src/caf/toolkit/__main__.py:65]: Added functionality for generating arguments from pydantic BaseModel's or our BaseConfig classes where optional config can be passed if using the latter. This hasn't been tested yet

**Comment by BenTaylor-TfN** [comment]
> I haven't had chance to write the documentation for this yet but I have implemented using the config. I still need to do some tests to make sure everything works but I think you should be able to run the translation from the CLI or by passing a config path, assuming I haven't broke anything with my recent changes.

Given that you're busy with some other work, is it worth focussing on getting the tests done for this, and add an issue for the docs for the future?
Getting the functionality out is probably more useful at first. If people use it, then we can use that experience to influence the docs too.

File: src/caf/toolkit/translation.py (hunk 2 above)
Comment (wsp-mbuckley) [reply src/caf/toolkit/translation.py:1295]: Added issue #109

**Comment by wsp-mbuckley** [comment]
Add issue about docs (#110) and tested both translations with config or command-line arguments.

**Review by isaac-tfn — APPROVED** [review]
All look good - might be worth a quick chat at some point as it looks like more has been added than just CLI stuff?
